crates/notion-core/src/session.rs

```diff
@@ -193,30 +191,27 @@ impl Session {
         self.toolchain.get_active_node().map(|ref nv| nv.clone())
     }
 
-    /// Fetches a version of Node matching the specified semantic verisoning
-    /// requirements.
-    pub fn fetch_node(&mut self, matching: &VersionSpec) -> Fallible<Fetched<NodeVersion>> {
-        let inventory = self.inventory.get_mut()?;
-        let config = self.config.get()?;
-        inventory.fetch_node(matching, config)
+    /// Installs a Tool matching the specified semantic versioning requirements,
+    /// and updates the `toolchain` as necessary.
+    pub fn install(&mut self, toolspec: &ToolSpec) -> Fallible<()> {
+        let distro_version = self.fetch(toolspec)?.into_version();
+        self.toolchain.set_active(distro_version)?;
+        Ok(())
     }
 
-    /// Sets the user toolchain's Node version to one matching the specified semantic versioning
-    /// requirements.
-    pub fn install_node(&mut self, matching: &VersionSpec) -> Fallible<()> {
+    /// Fetches a Tool version matching the specified semantic versioning requirements.
+    pub fn fetch(&mut self, tool: &ToolSpec) -> Fallible<Fetched<DistroVersion>> {
         let inventory = self.inventory.get_mut()?;
         let config = self.config.get()?;
-        let version = inventory.fetch_node(matching, config)?.into_version();
-        self.toolchain.set_active_node(version)?;
-        Ok(())
+        inventory.fetch(&tool, config)
     }
 
-    /// Updates toolchain in package.json with the Node version matching the specified semantic
+    /// Updates toolchain in package.json with the Tool version matching the specified semantic
     /// versioning requirements.
-    pub fn pin_node_version(&mut self, matching: &VersionSpec) -> Fallible<()> {
+    pub fn pin_tool(&mut self, toolspec: &ToolSpec) -> Fallible<()> {
```

Comment:
Would this one also read nicely as just `pin`?

Reply:
yup! fixed this as well